### PR TITLE
Stop overriding specs during double-checks

### PR DESCRIPTION
**Pull request by indirect**
This fixes my test-case reproduction of #6072. I’m pretty sure it’s because
the double-check was overriding the locally installed index specs with
specs fetched from the RubyGems API, causing Bundler to conclude that the
gems aren’t installed, even though they are.

@deivid-rodriguez @y-yagi can you verify that this patch fixes the issues with `inline` that you were seeing?

### CODE REVIEW

**Comment by y-yagi** [comment]
@indirect Thank you for fixing! I confirmed that this patch is solved my issue.

File: spec/runtime/inline_spec.rb [hunk 1]
```diff
@@ -267,4 +263,29 @@ def confirm(msg, newline = nil)
     expect(last_command).to be_success
     expect(last_command.stdout).to eq "1.0.0"
   end
+
+  describe "when path gems depend on API server gems" do
```
Comment (segiddins) [inline spec/runtime/inline_spec.rb]: this test doesn't seem to use `inline` at all?

File: spec/runtime/inline_spec.rb [hunk 2]
```diff
@@ -267,4 +263,29 @@ def confirm(msg, newline = nil)
...
+    let(:source_hostname) { "localgemserver.test" }
+    let(:source_uri) { "http://#{source_hostname}" }
+
+    before do
+      build_lib "needs_rake", "1.0.0" do |s|
```
Comment (segiddins) [inline spec/runtime/inline_spec.rb]: `needs_rack`?

File: spec/runtime/inline_spec.rb [hunk 3]
```diff
@@ -267,4 +263,29 @@ def confirm(msg, newline = nil)
...
+        s.add_dependency "rack", "= 1.0.0"
+      end
+
+      gemfile <<-G
+        source "#{source_uri}"
+        gem "needs_rake", path: "#{lib_path("needs_rake-1.0.0")}"
```
Comment (segiddins) [inline spec/runtime/inline_spec.rb]: hash rockets

**Comment by indirect** [comment]
@segiddins addressed your feedback, the test now does what I thought would trigger the bug... but the test passes. Any ideas about how to set up the right kind of double-check that will overwrite the index entry for the installed gem with an index entry for a gem from the API server?

**Comment by segiddins** [comment]
Are you saying the test doesn’t fail without your patch? Also, rubocop.

**Comment by indirect** [comment]
@segiddins fixed, and yes the test passes even without my patch. :(

**Comment by indirect** [comment]
Some progress... managed to reduce a very small repro case that still fails when run directly via `ruby`, over here: https://gist.github.com/indirect/e84b36fc854409916a569e53e2756deb

**Comment by indirect** [comment]
OKAY this is a working test. At least, it's red for me locally before the patch in this PR, and green for me after the patch in this PR. @segiddins r?

**Comment by segiddins** [comment]
This is fine, but I'd really like to get rid of the worldworld spec eventually

**Comment by indirect** [comment]
@segiddins I completely agree, but I think we need a better simulator of a source as complex as production rubygems to be able to do that. And I don’t want that to be a blocker for this bugfix. 👍🏻

**Comment by indirect** [comment]
@bundlerbot r+

**Comment by hsbt** [comment]
Can anyone retry this?

**Comment by segiddins** [comment]
@bundlerbot retry

**Pull request merged**